GuessTheNumber.java

```diff
@@ -20,5 +21,104 @@
  * "There are X players with better scores") or if he's at the top of the table.
  * Before the next game, the top 3 scores (which are greater than 0) will be printed
  */
+import java.util.Scanner;
+
 public class GuessTheNumber {
+	public static void main(String[] args) {
+		Scanner input = new Scanner(System.in);
+
+		System.out.print("Please enter number of digits for the game (2 to 10): ");
+		int targetNumberSize = input.nextInt();
+		while (targetNumberSize < 2 || targetNumberSize > 10) {
+			System.out.println("You can only enter value from 2 to 10. Please try again.");
+			targetNumberSize = input.nextInt();
+		}
+
+		int[] userGuessArray = new int[targetNumberSize];
+		int[] targetNumber = new int[targetNumberSize];
+		for (int i = 0; i < targetNumberSize; i++) {
+			targetNumber[i] = 10;
+		}
+
+		// Generate target number
+		for (int i = 0; i < targetNumberSize; i++) {
+			int digitToAdd = (int) (Math.random() * 10);
+			for (int j = 0; j < targetNumberSize; j++) {
+				if (digitToAdd == targetNumber[j] || (digitToAdd == 0 && j == 0)) {
+					digitToAdd = (int) (Math.random() * 10);
+					j = -1;
+				}
+			}
+			targetNumber[i] = digitToAdd;
+		}
+
+		// Start of game loop
+		boolean guessed = false;
+		int[] trials = new int[100];
+		int[] trialsCorrectPlaces = new int[100];
+		int[] trialsCorrectNumbers = new int[100];
+		int trialIndex = 0;
+
+		while (!guessed) {

```

Comment:
how about when the user enters more than 100 trials ?!

Reply:
I would write "Go back to school, you are too young for this game!" :)
Anyway, List.add() would do the job. Or int[2147483647]. Or long[9223372036854775807]. :)

Reply:
:)
limit the number of trials to say 100, and let the user know about this limit. once the user exceeds this limit, he looses